### ISSUE TITLE
SigLip2 text pooler output selection

### ISSUE THREAD

**Issue by Jason3900** [huggingface/transformers#39269]
### System Info

transformers==4.51.3
python==3.10.16
torch==2.7.0+cu128

### Who can help?

@amyeroberts @qubvel 

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Hey. I'm finetuning a siglip2 model and find some issues. 
From the original code from [google](https://github.com/google-research/big_vision/blob/0127fb6b337ee2a27bf4e54dea79cff176527356/big_vision/models/proj/image_text/text_transformer.py#L83), the text ids' last position is assumed to be eos token. While if I use transformers preprocessor following the doc, text whose token length is less than max_length will pad to max_length. In that case, the last token is not eos but pad. Since the [code](https://github.com/huggingface/transformers/blob/17b3c96c00cd8421bff85282aec32422bdfebd31/src/transformers/models/siglip2/modeling_siglip2.py#L692) always takes the last position to feed into the pool layer, I think there might be some mismatches?  
```
        # Assuming "sticky" EOS tokenization, last token is always EOS.
        pooled_output = last_hidden_state[:, -1, :]
        pooled_output = self.head(pooled_output)
```
And another question is the preprocessor's output doesn't contain attention_mask, should I manually build it according to padding positions or the original model doesn't use attention_mask for padding at all?

### Expected behavior

I want some explanation.

**Comment by qubvel** [comment]
Hey @Jason3900, thanks for opening the issue. SIglip and Siglip2 models were initially trained **without** an attention mask to mask text padding tokens, that's why the processor doesn't create it.

Regarding

```
        # Assuming "sticky" EOS tokenization, last token is always EOS.
        pooled_output = last_hidden_state[:, -1, :]
        pooled_output = self.head(pooled_output)
```

As far as I recall, I paid attention to it while converting the model, and the implementation should be OK, because we benchmarked converted Siglip2 on Imagenet and got the same score as the original implementation (as well as getting the same logits on dummy examples). It might be just a misleading comment

**Comment by Jason3900** [comment]
@qubvel Thanks a lot! To be clear, actually it should use the last token's hidden states to feed into the head layer regardless whether it's a pad or eos, right?

**Comment by qubvel** [comment]
> To be clear, actually it should use the last token's hidden states to feed into the head layer regardless whether it's a pad or eos, right?

Yes.

Feel free to submit a PR to remove the comment if you have bandwidth 🤗

**Comment by qubvel** [comment]
Fixed in https://github.com/huggingface/transformers/pull/39378
